mongodbatlas/resource_mongodbatlas_organization.go

```diff
@@ -0,0 +1,178 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mwielbut/pointy"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+func resourceMongoDBAtlasOrganization() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceMongoDBAtlasOrganizationCreate,
+		ReadContext:   resourceMongoDBAtlasOrganizationRead,
+		UpdateContext: resourceMongoDBAtlasOrganizationUpdate,
+		DeleteContext: resourceMongoDBAtlasOrganizationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMongoDBAtlasOrganizationImportState,
+		},
+		Schema: map[string]*schema.Schema{
+			"org_owner_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"org_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"public_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"private_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"role_names": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceMongoDBAtlasOrganizationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*MongoDBClient).Atlas
+
+	createRequest := new(matlas.CreateOrganizationRequest)
+	apiKey := &matlas.APIKeyInput{}
+
+	createRequest.Name = d.Get("name").(string)
+	createRequest.OrgOwnerID = pointy.String(d.Get("org_owner_id").(string))
+	apiKey.Desc = d.Get("description").(string)
+	apiKey.Roles = expandStringList(d.Get("role_names").(*schema.Set).List())
+
+	organization, resp, err := conn.Organizations.Create(ctx, &matlas.CreateOrganizationRequest{
+		Name:       createRequest.Name,
+		OrgOwnerID: createRequest.OrgOwnerID,
+		APIKey:     apiKey,
+	})
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(fmt.Errorf("error create Organization: %s", err))
+	}
+
+	if err := d.Set("private_key", organization.APIKey.PrivateKey); err != nil {
+		return diag.FromErr(fmt.Errorf("error setting `public_key`: %s", err))
```

Comment:
should this be private_key? 
```suggestion
		return diag.FromErr(fmt.Errorf("error setting `private_key`: %s", err))
```

Reply:
corrected